subsys/trusted_storage/src/aead/aead_crypt_mbedtls_chachapoly.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/random/rand32.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <mbedtls/chachapoly.h>
+
+#include "settings_trust.h"
+
+psa_status_t trusted_storage_aead_init(void)
+{
+	return PSA_SUCCESS;
+}
+
+size_t trusted_storage_aead_get_encrypted_size(size_t data_size)
+{
+	/* 16 bytes tag followed by encrypted data */
+	return 16 + data_size;
+}
+
+psa_status_t trusted_storage_aead_encrypt(const void *key_buf, size_t key_len,
+					  const void *nonce_buf, size_t nonce_len,
+					  const void *add_buf, size_t add_len,
+					  const void *input_buf, size_t input_len, void *output_buf,
+					  size_t output_size, size_t *output_len)
+{
+	mbedtls_chachapoly_context ctx;
```

Comment:
Why do we need an mbedtls backend for aead?

Reply:
The idea is to try to upstream that and I got told therefore we should have an mbedtls option, as there might be the risk that they don't what to use psa api. As ITS will be called as part of other PSA calls.
@Vge0rge correct me if I'm wrong, I remember disscussing that with you.